src/lib/support/CHIPMemString.h

```diff
@@ -21,31 +21,79 @@
  *      This file defines string operations that allocate heap memory.
  */
 
-#ifndef CHIP_MEM_STRING_H
-#define CHIP_MEM_STRING_H
+#pragma once
 
 #include <stdlib.h>
+#include <string.h>
+
+#include <support/ScopedBuffer.h>
 
 namespace chip {
 namespace Platform {
 
+/**
+ * Copies a C-style string.
+ *
+ * This differs from `strncpy()` in some important ways:
+ *  - `dest` can be nullptr, in which case no copy is attempted, and the function returns nullptr.
+ *  - A non-nullptr result is always null-terminated.
+ *
+ * @param[in]  dest             Destination string buffer (which must be at least `length`+1 bytes)
+ *                              or nullptr.
+ *
+ * @param[in]  source           String to be copied.
+ *
+ * @param[in]  length           Length to be copied.
+ *
+ * @retval  Same as `dest`.
+ */
+inline char * CopyString(char * dest, const char * source, size_t length)
+{
+    if (dest)
+    {
+        memcpy(dest, source, length);
```

Comment:
any issues if source is shorter than length? Maybe we could use strncpy internally.

Reply:
I was about to say no (NUL in the source is fine, NUL not in the source is fine) but there is one case where memcpy would break (NUL-terminated source closely followed by dangerous addresses).